src/sql/workbench/contrib/notebook/browser/notebookActions.ts

```diff
@@ -49,18 +50,27 @@ export class AddCellAction extends Action {
 	) {
 		super(id, label, cssClass);
 	}
-	public async run(context: INotebookEditor): Promise<any> {
-		//Add Cell after current selected cell.
+	public async run(context: INotebookEditor | CellContext): Promise<any> {
 		let index = 0;
-		if (context && context.cells) {
-			let notebookcomponent = context as NotebookComponent;
-			let id = notebookcomponent.activeCellId;
-			if (id) {
-				index = context.cells.findIndex(cell => cell.id === id);
-				index = index + 1;
+		if (context instanceof CellContext) {
+			if (context && context.model && context.model.cells) {
```

Comment:
nit - but these kind of checks can be made simpler using null-coalescing `if (context.model?.cells) {`

(you can get rid of the first check since undefined will never be a CellContext)